tests/Functional/Platform/DateExpressionTest.php

```diff
@@ -30,19 +31,21 @@ public function testDifference(string $date1, string $date2, int $expected): voi
         self::assertEquals($expected, $diff);
     }
 
-    /** @return string[][]|int[][] */
+    /** @return array<string, array{string, string, int}> */
     public static function differenceProvider(): iterable
     {
-        $date1    = new DateTimeImmutable();
-        $date2    = new DateTimeImmutable('2018-04-10 10:10:10');
-        $expected = $date1->modify('midnight')->diff(
-            $date2->modify('midnight'),
-        )->days;
+        $datetime1 = new DateTimeImmutable();
+        $datetime2 = new DateTimeImmutable('2018-04-10 10:10:10');
+        $date1     = $datetime1->modify('midnight');
+        $date2     = $datetime2->modify('midnight');
+        assert($date1 !== false && $date2 !== false);
```

Comment:
Should these be two independent PHPUnit assertions?

Reply:
test framework assertions are supposed to be used when it's related to the use case that's the focus of this test. If `$date1` and `$date2` are false we have a different and much greater problem than a failing date-difference test.

Reply:
I tend to use `assert` for assertions that are not really part of the test assertions, but a help for static analysis. I suppose in case of failure, the UX is going to be better with PHPUnit asserts though.

Reply:
The fact that these dates are not false is a pre-condition of the test itself which the static analyzer forces us to use. We know that it will fail if those are not met, so it's part of the test. We have a ton of such assertions in the codebase.

Reply:
My point is: it won't indicate a failure in the SUT.

Reply:
In this specific case, the failure would happen in the data provider, so technically, it's the job of PHPUnit to report it as a setup failure rather than a failure in the SUT.